packages/product/src/repositories/product-tag.ts

```diff
@@ -24,7 +24,7 @@ export class ProductTagRepository extends AbstractBaseRepository<ProductTag> {
     context: Context = {}
   ): Promise<ProductTag[]> {
     const manager = (context.transactionManager ??
-      this.manager_) as SqlEntityManager
+      this.manager_.fork()) as SqlEntityManager
```

Comment:
thought: Should we create an `activeManager_` getter, as we have in the core?

I feel like this is prone to errors - especially the `fork()`. One could easily forget to do this.

I don't know if we can create a getter in the pure sense because we need to check the context, but maybe a function:
```
this.activeManager(context)
```

Or something similar. 

What do you think?

cc @adrien2p

Reply:
What I did here is incorrect, we shouldn't be forking at a repository level. I just added it to confirm if this was truly the root cause of the failed tests.

By doing something like this, we clear the identity map in every respository method call and we lose the benefits of the identity map. We need the identity map scoped to a "request", in our case a service method call, so we should be ideally passing down a forked manager from the entry point methods. 

Ref: https://mikro-orm.io/docs/identity-map#-requestcontext-helper

Reply:
But yes, the helper method to get the active manager from context is a good idea.

Reply:
The InjectTransactionManager does that, it forks the manager before passing it down. So for read write this is good. For read only, I think the BaseRepository can have a get activeManager(context) and either return the transactionManger or a manager.fork() like you suggested 👍 in that case you need to update the repositoryservice interface to add that to the contract

Reply:
You can do that in the base branch if it is not merged yet 🙏